### PR TITLE
RFP Proposal: Chainlink x Filecoin: Use Filecoin for Ethereum zk-Rollup storage

### PR DESCRIPTION
**Pull request by weikengchen**
Today, the dominating cost of zk-Rollup is related to data availability. Putting data on layer-1, which is the standard solution for data availability, is very expensive. Several companies working in zk-Rollup have been thinking about their own ways of escaping from storing data on Ethereum (e.g., zkPorter, https://blog.matter-labs.io/zkporter-a-breakthrough-in-l2-scaling-ed5e48842fbf).

Given that Chainlink x Filecoin is going to provide an external adapter, assuming at least there is a data feed for the latest block hash or state tree hash, it is possible to enable running a light client inside the zkSNARK circuit, and therefore, the layer-2 can prove to everyone that the data has been reliably stored on Filecoin. 

This project is to explore the technical designs for it. The goal is to enable a zk-Rollup layer-2 to prove that certain data has been stored on Filecoin. The design needs to be efficient, which involves the technical details of zkSNARK and Filecoin's design. Note that Filecoin is using a number of cryptographic primitives that could make the light client in zkSNARK challenging. They will be the focus of this research.

In summary, we want to check if the external adapter would be sufficient, or needs changes, for proving Filecoin storage inside zkSNARK.

---

This project depends heavily on the external adapter that Chainlink and Filecoin are working on, and would need more information about the progress.

### CODE REVIEW

**Comment by momack2** [comment]
Very supportive of this RFP! The one question I have would be whether it can happen on a faster timeline, or if we can ensure each milestone deliverable is progressively published to the community as it's completed every 2 weeks  - since I think this area is of immediate interest to many groups who would want to absorb and iterate on the thinking here.

**Comment by anorth** [comment]
Love it. If granted, please reach out if you need help understanding Filecoin protocol.

cc @nicola

**Comment by weikengchen** [comment]
Dear all, really thanks for the support. 

re: the current status of the Chainlink x Filecoin API, based on the Chainlink x Filecoin Slack channel, I think at this moment it is an open space to explore, so it is no longer a concern of the project.

A faster timeline is doable as I can get more people (which may need more grant, as it is no longer a personal project, will follow up on it). Indeed, most of the ideas have been fleshed out, so it is mostly an effort of 

You should really approve this grant. It is only 7k. It should be a no-brainer even if it fails. I am in the startup space, this idea clearly has a huge amount of commercial values, and I would approve it today! Several decentralized storage companies have been thinking about this direction. Filecoin has competitors, such as Arweave. This is very useful to the Filecoin ecosystem and can boost the Filecoin token price in short terms and long terms.

This idea is not just planned as a dev grant. We should work on this!

**Comment by realChainLife** [comment]
Hi @weikengchen Thank you for this proposal. We would like to fund the work outlined here, please email devgrants@fil.org to discuss next steps.
